gnucash/report/trep-engine.scm

```diff
@@ -1676,7 +1827,19 @@ be excluded from periodic reporting.")
 
     (define (add-split-row split cell-calculators row-style transaction-row?)
       (let* ((account (xaccSplitGetAccount split))
-             (reversible-account? (acc-reverse? account)))
+             (reversible-account? (acc-reverse? account))
+             ;; Returns the number of required arguments of a function
+             (fn-arity (lambda (fn) (car (procedure-minimum-arity fn))))
+             ;; Calls the calculator-function for the provided cell with either
+             ;; two arguments (calculator-function split transaction-row?) or only
+             ;; one argument  (calculator-function split) for legacy support.
+             (calc-fn (lambda (cell split transaction-row?)
+                              (if (= (fn-arity (vector-ref cell 1)) 1)
+                                  ((vector-ref cell 1) split)
+                                  ((vector-ref cell 1) split transaction-row?))))
+             ;; Do any of the displayed columns have a subtotal flag?
+             (any-col-with-subtotals?
+               (< 0 (length(filter cell-with-subtotals? cell-calculators)))))
```

Comment:
We don't like `length` - it is O(N) rather than O(1). This one I think can be optimized to `(any cell-with-subtotals? cell-calculators)` which is still O(N) but can terminate early.

Reply:
In this case - I don't think it's necessary to define it here -- it's best to inline it into line 1867 so that the list traversal is only performed when necessary.

```scheme
                           (if (and opt-use-links? (or (cell-with-subtotals? cell)
                                                       (not (any cell-with-subtotals? cell-calculators))))
```

Reply:
Nice one! And funny too because I even wrote the any in my original code (as part of the function name). It's like I was 99% there but couldn't quite make it. Will update.